swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/images/AmazonImageHandler.kt

```diff
@@ -84,101 +86,90 @@ class AmazonImageHandler(
               "credentials" to workConfiguration.account.name,
               "imageIds" to markedResources.map { it.resourceId }.toSet(),
               "cloudProvider" to AWS,
-              "region" to workConfiguration.location
+              "region" to workConfiguration.location,
+              "stageTimeoutMs" to "900000" // fifteen minutes
             )
           )
         ),
         description = "Deleting Images :${markedResources.map { it.resourceId }}"
       )
     ).let { taskResponse ->
-      val taskId = parseTaskId(taskResponse)
-      val taskStatus = getTask(taskId).status
-      when {
-        taskStatus.isSuccess() -> {
-          log.info("Orca Task: {} successfully deleted {} images: {}", taskId, markedResources.size, markedResources)
-          markedResources.forEach {
-            send(it)
-          }
-        }
-        taskStatus.isFailure() -> throw IncompleteOrFailedDeletionException("Failed to delete $markedResources in orca task $taskId")
-        taskStatus.isIncomplete() -> {
-          val deleteChannel = monitorDelete(taskId, markedResources, workConfiguration)
-          deleteChannel.consumeEach { send(it) }
-        }
-      }
+      taskTrackingRepository.add(
```

Comment:
The OrcaTaskMonitoringAgent now periodically monitors these tasks, and emits spring events when they're finished.